### PR TITLE
Add Session Class UML

### PR DESCRIPTION
**Pull request by ldinghan**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -157,6 +157,37 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+* `SessionStudents`: An essential component of every session is the list of students participating. The SessionStudents class keeps track of the students present in a particular session. This class allows for efficient management of attendance records and plays a vital role in generating attendance reports.
+
+* `SessionRemark`: Sometimes, additional information about a session is necessary, such as special instructions, topics covered, or any other relevant remarks. The SessionRemark field offers the flexibility to include such notes.
+
+The Session class also provides a set of getter methods that enable access to these attributes. For instance, you can retrieve the session number, list of students, or session remarks using these methods.
+
+Below is a class diagram describing the implementation of `Session` and its respective fields.
+
+###### NEED TO ADD CLASS DIAGRAM
```
Comment (wesho1107) [inline docs/DeveloperGuide.md]: Probably to exclude placeholder as the UML Diagram is already present

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -157,6 +157,37 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+**Alternative 1 (current choice):** The `CreateSession` feature takes in arguments of varying number of student names.
+- Pros:This choice offers a high degree of flexibility when it comes to adding students to a session. You can add any number of students when creating a session, which is essential for accommodating different class sizes and situations.
+**Alternative 2:** An alternative design could involve creating two separate constructor methods within the `CreateSession` class. One constructor would be responsible for adding a student to an existing `SessionStudents`, and the other would take an entire `SessionStudents` object as an argument.
+- Cons: This alternative introduces additional complexity in terms of validating user input and checking for null values upon execution. It may be less intuitive than the current approach. 
+
+By opting for the current choice (Alternative 1), the implementation remains straightforward and user-friendly, allowing for versatile usage scenarios. It ensures that users can efficiently create sessions and add students to them without unnecessary constraints or complications.
```
Comment (wesho1107) [inline docs/DeveloperGuide.md:185]: Good! But I'm unsure if required, will double check

File: docs/diagrams/SessionClassUML.puml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+class SessionRemark {
+  + sessionRemark: String
+  + <u>MESSAGE_CONSTRAINTS: String
+  + <u>VALIDATION_REGEX: String
+  + <u>isValidSessionRemark(String): boolean
+  + hashCode(): int
+  + toString(): String
+}
```
Comment (wesho1107) [inline docs/diagrams/SessionClassUML.puml]: Is toString() and hashCode() necessary? Omit less important details and repetitive details.

**Review by wesho1107 — APPROVED** [review]
Other than a few suggestions, LGTM!

File: docs/diagrams/SessionClassUML.puml (hunk 3 above)
Comment (ldinghan) [reply docs/diagrams/SessionClassUML.puml]: Will do! Thanks for the suggestion

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ldinghan) [reply docs/DeveloperGuide.md]: Nice catch! Owe you one :)

**Pull request merged**